File: Sort Colors/sol2.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        int arr[3] = {0,0,0};
        for(int i=0; i < nums.size() ; i++)
        {
            if(nums[i] == 0)
                arr[0]+=1;
            else if(nums[i] == 1)
                arr[1]+=1;
            else if(nums[i] == 2)
                arr[2]+=1;
        }
        for(int j=0;j < arr[0]; j++)
            nums[j] = 0;
        for(int j=arr[0];j < arr[0] + arr[1];j++)
            nums[j] = 1;
        for(int j=arr[0]+arr[1];j < nums.size();j++)
            nums[j] = 2;
    }

};

File: Find the Duplicate Number/sol2.py
class Solution:
    def findDuplicate(self, nums: List[int]) -> int:
        #consider the value of the array as pointer, where they point to nodes at position specified by value
        slow, fast = 0, 0
        while True:
            slow = nums[slow]
            fast = nums[nums[fast]]
            if slow == fast:#intersection
                break
        slow2 = 0
        while True:
            slow2 = nums[slow2]
            slow = nums[slow]
            if slow ==slow2:
                break
        return slow